syncing/src/main/kotlin/maru/syncing/beaconchain/pipeline/DownloadBlocksStep.kt

```diff
@@ -23,8 +23,32 @@ import tech.pegasys.teku.networking.eth2.rpc.core.RpcException
 import tech.pegasys.teku.networking.p2p.peer.DisconnectReason
 import tech.pegasys.teku.networking.p2p.reputation.ReputationAdjustment
 
-class DownloadBlocksStep(
+interface DownloadPeerProvider {
+  fun getDownloadingPeer(targetBlockNumber: ULong): MaruPeer
```

Comment:
```suggestion
  fun getDownloadingPeer(downloadRangeEndBlockNumber: ULong): MaruPeer
```

This helps to understand what his value is semantically. Initially, I thought this was the head sync target.